src/build_workflow/build_incremental.py

```diff
@@ -0,0 +1,64 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+from typing import List
+
+from manifests.build_manifest import BuildManifest
+from manifests.input_manifest import InputManifest
+
+
+class BuildIncremental:
+    def __init__(self, input_manifest: InputManifest, distribution: str):
+        self.distribution = distribution
+        self.input_manifest = input_manifest
+
+    # Given input manifest and return a list of what components changed and added.
+    def commits_diff(self, input_manifest: InputManifest) -> List[str]:
+        build_manifest_path = os.path.join(self.distribution, "builds", input_manifest.build.filename, "manifest.yml")
```

Comment:
I am assuming we will be downloading the build-manifest and storing it in the path mentioned before this logic executes?